velox/vector/arrow/Bridge.cpp

```diff
@@ -35,9 +35,25 @@ struct VeloxToArrowBridgeHolder {
   const void* buffers[kMaxBuffers];
 };
 
-// Main release function. Arrow standard requires it to recurse down to children
-// and dictionary arrays, and set release and private_data to null to signal it
-// has been released.
+// Structure that will hold buffers needed by ArrowSchema. This is opaquely
+// carried by ArrowSchema.private_data
+struct VeloxToArrowSchemaBridgeHolder {
+  // Unfortunately, we need two vectors here since ArrowSchema takes a
+  // ArrowSchema** pointer for children (so we can't just cast the
+  // vector<unique_ptr<>>), but we also need a member to control the
+  // lifetime of the children objects.
+  std::vector<ArrowSchema*> childrenRaw;
```

Comment:
nit: shall we have a comment state the contract that `childrenRaw[i] == childrenOwned[i].get()`?